### ISSUE TITLE
[Bug] Broken function: operands of ICmp instruction not of the same type

### ISSUE THREAD

**Issue by RogueMacro** [beefytech/Beef#950]
Crash dump in zip shows error in more detail.

[Project.zip](https://github.com/beefytech/Beef/files/6086251/Project.zip)

**Comment by kallisto56** [comment]
I believe this issue may be closed.
TLDR: On latest nightly (https://github.com/beefytech/Beef/commit/1805316b70387a8cfabdee45f1e5204fa0c5d649) it builds and runs fine.

---
Crash dump was related to `enum KeyCode.IsLetter`:
```
Broken function found, compilation aborted! in C:\Jenkins\workspace\Beef\IDEHelper\DebugManager.cpp:578
Both operands to ICmp instruction are not of the same type!
  %0 = icmp sge i64 %this, i8 97, !dbg !176
Both operands to ICmp instruction are not of the same type!
  %1 = icmp sle i64 %this, i8 122
Both operands to ICmp instruction are not of the same type!
  %2 = icmp sge i64 %this, i8 65, !dbg !176
Both operands to ICmp instruction are not of the same type!
  %3 = icmp sle i64 %this, i8 90, !dbg !176
in function ?get__IsLetter__im@1$KeyCode@UI@Steak@bf@@SA_N_J@Z
```

Crash dump was made in:
```
File Version: 0.43.0 (Nightly 02/27/2021)
Product Version: 4feda1da592f4848bca8679a8413ac1dd30822d3
```

Starting from stable v0.43.0 and onwards, this project builds and runs with additional checks to `KeyCode` without any problems.

In case if you want to test it yourself on the latest build: [Steak.UI.zip](https://github.com/user-attachments/files/18938895/Steak.UI.zip)

- Open workspace at `Steak.UI\Steak.UI.Tests\PluginSource` first and build it (it's a collection of plugins as dlls);
- Open workspace at root folder (`Steak.UI`) and start debugging;

Changes:
minor SDL2 adjustments:
```C#
// From this:
int32 x = 0, y = 0;
SDL.GetMouseState(&x, &y);

// To this:
SDL.GetMouseState(var x, var y); // GetMouseState changed signature at one point
```

and a small fix to dll-loading:
```C#
// From this:
Window.HModule module = Windows.LoadLibraryA(path.Ptr);

// To this:
var module = Windows.LoadLibraryA(path.ToScopeCStr!());
```
I have no idea why I wrote this wall of text.
